### PR TITLE
Feature/parcel and discord jumps

### PR DESCRIPTION
**Pull request by tescher**
Support for multiple currencies, and associated contract addresses. The Parcel.money export needs the contract addresses if it's not ETH.

Also modified the Parcel.money export to exclude bounties that are not paid in ETH or in tokens that have contract addresses, as well as bounties whose claimants don't have a registered wallet address.

### CODE REVIEW

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 1]
```diff
@@ -201,31 +190,46 @@ const SelectExport = ({
...
-								if (bounty['claimedBy']) {
-									bounty['compositeName'] = bounty['claimedBy']['discordHandle'];
-								}
-								bounty['compositeName'] += ' - ' + bounty['title'];
-								return bounty;
-							});
+							if (bounties) {
+								const exportBounties = (bounties
+									.filter(({ _id }) => selectedBounties.includes(_id))
+									.filter(bounty => bounty.claimedBy && bounty.payeeData.walletAddress && ALLOWED_CURRENCY_CONTRACTS[bounty.reward.currency as ALLOWED_CURRENCIES_TYPE]) as any[])
```
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/index.tsx:196]: Something about this line doesn't sit right with me. Can you explain it?

Rationale: you're filtering out bounties that don't contain a `claimedBy` `walletAddress` and where the reward currency is incorrect - it sounds to me like there are some validations that need to happen elsewhere?

Then casting to `any` just strikes me as another sign that this filter is doing some of the legwork that another part of the codebase should be handling.

Definitely not a crisis - if it works then cool. But just something to consider

File: packages/react-app/src/constants/currency.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export const ALLOWED_CURRENCY_CONTRACTS = {
+	'BANK': '0x2d94AA3e47d9D5024503Ca8491fcE9A2fB4DA198',
+	'ETH': 'ETH',
+	'USDC': '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+	'USDT': '0xdAC17F958D2ee523a2206206994597C13D831ec7',
+	'BTC': '',
+};
+
+export const ALLOWED_CURRENCIES = Object.keys(ALLOWED_CURRENCY_CONTRACTS);
+export type ALLOWED_CURRENCIES_TYPE = ObjectKeys<typeof ALLOWED_CURRENCY_CONTRACTS>;
```
Comment (jordaniza) [inline packages/react-app/src/constants/currency.ts:12]: An alternative approach could be to define a union type for `ALLOWED_CURRENCIES = x | y | z`

You can then define `ALLOWED_CURRENCY_CONTRACTS: Record<ALLOWED_CURRENCIES | string> = {}`

I believe react hook form is smart enough to work with the vanilla record type but you could also add this to the model for the bounty. Would save you having to cast

```ts
bounty.reward.currency = ALLOWED_CURRENCY_CONTRACTS[bounty.reward.currency as ALLOWED_CURRENCIES_TYPE];
```

**Review by jordaniza — APPROVED** [review]
Suggestions for review but no blockers

File: packages/react-app/src/components/pages/Bounties/index.tsx (hunk 1 above)
Comment (tescher) [reply packages/react-app/src/components/pages/Bounties/index.tsx:196]: The filter only applies on this particular export format - we don't want any Parcel.money fields missing data. Missing wallet address or having a bounty paid in BTC is fine, just not for bounties exported to Parcel

**Pull request merged**